rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3FilterHandler.java

```diff
@@ -67,15 +72,41 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request)
         // the request will be inmediately rejected since it won't be parsed as a valid URI
         // and won't work to match an item on rpc.host
         String hostHeader = headers.get(HttpHeaders.Names.HOST);
-        String parsedHeader = parseHostHeader(hostHeader);
 
-        if (!acceptedHosts.contains(parsedHeader)) {
-            logger.debug("Invalid header HOST {}", hostHeader);
-            response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_REQUEST);
-            ctx.write(response).addListener(ChannelFutureListener.CLOSE);
+        if (hostHeader.isEmpty()) {
+            this.serveRequest(ctx, request);
+            return;
+        }
+
+        String host = parseHostHeader(hostHeader);
+
+        if (isValidIpAddress(host)) {
```

Comment:
could you covert this with unit test?

also do you think we could move this check to the end - just after checks of `acceptedHosts`? cause `acceptedHosts.contains()` seems less resource-intensive and the case when a host is an ip shouldn't be that common

Reply:
Done